lookout/core/data_requests.py

```diff
@@ -88,6 +89,66 @@ def _get_channel(self) -> grpc.Channel:
         return channel
 
 
+class UnicodeDataService(DataService):
```

Comment:
So I am actually against this class. We cannot say that all analyzers require Unicode data. Those which need Unicode should use the provided decorators.

Reply:
> We cannot say that all analyzers require Unicode data.

We do not say that. 
the code in style-analyzer will be:
```python
    @with_unicode_data_service
    @with_changed_uasts_and_contents
    def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
                data_service: UnicodeDataService, changes: Iterator[Change], **data) -> List[Comment]:
```

And the abstract `Analyzer` class is the same. If you do not need unicode data you just have:
```python
    @with_changed_uasts_and_contents
    def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
                data_service: UnicodeDataService, changes: Iterator[Change], **data) -> List[Comment]:
```

So you just need a decorator. You do not deal with this class directly. 

Unfortunately, we cannot avoid having this class at all because if you request the new UAST for the file (like we do in the `UASTStabilityChecker` class) you expect it to be in a Unicode format. So I do not see another way to do so.

Reply:
Our analyzers are not supposed to work directly with responses, but rather through the decorators, so that's fine. This class should be removed. We should never unicodify anything automatically.

Reply:
So we should add an argument to `with_changed_uasts_and_contents` and the family to unicodify the nodes:

```python
@with_changed_uasts_and_contents(unicode=True)
def analyze(...):
```

Changing the data service itself is way too magical. It changes the method signature, instead of merely appending new arguments.

Reply:
Do you agree that when you have two different data formats (byte from `DataService` and Unicode from `Files`/`Changes` ) is it knotty and can lead to bugs?

I see only two options.
1. We keep `UnicodeDataService`
2. We do not include any transformation into decorator and do it manually in the function. So no confuses, only one (byte) format and more straightforward approach. Plus no magic for free :)

I'd like 1st option. A signature is fine because `UnicodeDataService` is an inheritor of `DataService`. 

But I do not like converting `Files`/`Changes` and keeping `DataService`

The choice is yours. :)

Reply:
My choice is to remove `UnicodeDataService` and implement `unicode=True` in the existing decorators.

Reply:
We decided in DM that once I have more time for consideration, we should discuss `UnicodeDataService` in more detail.